### PR TITLE
Check provider constraints

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/15852

### CODE REVIEW

File: pkg/validation/app_test.go [hunk 1]
```diff
@@ -662,6 +664,33 @@ func Test_ValidateMetadataConstraints(t *testing.T) {
...
+					Name:      "giantswarm-kiam-1.4.0",
+					Namespace: metav1.NamespaceDefault,
+				},
+				Spec: v1alpha1.AppCatalogEntrySpec{
+					Restrictions: &v1alpha1.AppCatalogEntrySpecRestrictions{
+						CompatibleProviders: []v1alpha1.Provider{"azure"},
+					},
+				},
+			},
+			expectedErr: "validation error: app `kiam` can only be installed in [`azure`] provider",
```
Comment (tomahawk28) [inline pkg/validation/app_test.go:692]: Well, it's totally different actually, but why not.

File: pkg/validation/app.go [hunk 2]
```diff
@@ -147,6 +147,13 @@ func (v *Validator) validateMetadataConstraints(ctx context.Context, cr v1alpha1
 		return nil
 	}
 
+	if len(entry.Spec.Restrictions.CompatibleProviders) > 0 {
+		if !contains(entry.Spec.Restrictions.CompatibleProviders, v1alpha1.Provider(v.provider)) {
+			return microerror.Maskf(validationError, "app %#q can only be installed in %#q providers",
+				cr.Spec.Name, entry.Spec.Restrictions.CompatibleProviders)
```
Comment (rossf7) [inline pkg/validation/app.go:153]:
```suggestion
			return microerror.Maskf(validationError, "app %#q can only be installed for providers %#q not %#q",
				cr.Spec.Name, entry.Spec.Restrictions.CompatibleProviders, v.provider)
```

What about this for the error?

**Review by rossf7 — COMMENTED** [review]
Looks good just a question on the error message.

File: pkg/validation/app.go (hunk 2 above)
Comment (tomahawk28) [reply pkg/validation/app.go]: Much better, I like it.

**Review by rossf7 — APPROVED** [review]
Nice, LGTM

**Pull request merged**